db/system_distributed_keyspace.cc

```diff
@@ -90,6 +90,37 @@ schema_ptr cdc_generations() {
     return schema;
 }
 
+/* An internal table used by nodes to exchange CDC generation data. */
+schema_ptr cdc_generations_v2() {
+    thread_local auto schema = [] {
+        auto id = generate_legacy_id(system_distributed_keyspace::NAME_V2, system_distributed_keyspace::CDC_GENERATIONS_V2);
+        return schema_builder(system_distributed_keyspace::NAME_V2, system_distributed_keyspace::CDC_GENERATIONS_V2, {id})
+                /* The unique identifier of this generation. */
+                .with_column("id", uuid_type, column_kind::partition_key)
+                /* The generation describes a mapping from all tokens in the token ring to a set of stream IDs.
+                 * This mapping is built from a bunch of smaller mappings, each describing how tokens in a subrange
+                 * of the token ring are mapped to stream IDs; these subranges together cover the entire token ring.
+                 * Each such range-local mapping is represented by a row of this table.
+                 * The clustering key of the row is the end of the range being described by this row.
+                 * The start of this range is the range_end of the previous row (in the clustering order, which is the integer order)
+                 * or of the last row of this partition if this is the first the first row. */
+                .with_column("range_end", long_type, column_kind::clustering_key)
+                /* The set of streams mapped to in this range. */
+                .with_column("streams", cdc_streams_set_type)
+                /* The value of the `ignore_msb` sharding parameter of the node which was the owner of this token range
+                 * when the generation was first created. Together with the set of streams above it fully describes
+                 * the mapping for this particular range. */
+                .with_column("ignore_msb", byte_type)
```

Comment:
This solves the problem of a node being replaced with a machine that has different number of CPUs, right? Can we retire IGNORE_MSB from gossip now?

Reply:
NVM. It does not solve the CPU problem but maybe we could add a column with number of shards and that way fix the CPU problem?

Reply:
Actually length of streams is the number of CPUs so all's good.

Reply:
`IGNORE_MSB` in gossip is used when creating new CDC generations so we can't remove it (unless we introduce a new method of obtaining this parameter).